parser/src/parsers.rs

```diff
@@ -569,7 +569,10 @@ pub fn func_def(input: Cursor) -> ParseResult<Spanned<ContractStmt>> {
 }
 
 pub fn arg_list(input: Cursor) -> ParseResult<Vec<Spanned<FuncDefArg>>> {
-    separated(arg_def, op(","), true)(input)
+    match input[0] {
+        Token { string: ")", .. } => Ok((input, vec![])),
```

Comment:
@g-r-a-n-t  not sure if this is the intended way of handling it?

Reply:
If it's the case that `separated` should not be able handle an empty list, I think this is ok. But, it seems odd to me. Maybe this is not the intended behavior of `separated` and we should fix that instead?

I tried changing [this line](https://github.com/ethereum/fe/blob/5b9d302cef06595c6c76486415c6a109e899cb99/parser/src/builders.rs#L278) to

```rust
let (input, first) = match parser(input) {
    Ok(result) => result,
    Err(_) => return Ok((input, vec![]))
};
```

which I thought would allow it to handle empty lists, but it ended breaking a bunch of other stuff, so I'm not quite sure.

Reply:
>I tried changing this line to

Wouldn't that swallow legitimate errors by treating them as an empty argument list?

My instinct says it isn't too bad to let `separated`  continue to work on the assumption of dealing with at least one item and handle allowed empty cases explicitly. Another possible idea would be to introduce another parameter `allow_empty` to `separated` to manage this. My intuition says we could leave it as is and refactor it if this pops up more often. What do you think?

Reply:
I'm alright with this